### DISCUSSION TITLE
Iterating through an array of objects

### DISCUSSION THREAD

**Discussion by mayank1608** [nlohmann/json#2524] [Q&A] [answered]
Is there a way to iterate Array of Objects??
I have tried this way but not working:

```
const color = [ { color: "red", value: "#f00" }, { color: "green", value: "#0f0" }, { color: "blue", value: "#00f" } ]
json jsonObj;
json a = json::array();
jsonObj = json::parse(color);
for (json::iterator it = jsonObj.begin(); it != jsonObj.end(); ++it) {
	// for (auto& el : *it.items()) {
	// 	std::cout << el.key() << " : " << el.value() << "\n";
	// }
	a.push_back(it.key());	// returns index of jsonObj
	std::cout << it.key() << '\n';
}

std::cout << a << '\n';
```

Thanks in advance.

**Comment by nlohmann** [comment]
What do you want to achieve? Using iterators as well as the `items` functions is the way to go. But I don't understand where you struggle.

**Comment by mayank1608** [comment, marked as answer]
Hi @nlohmann sorry for the late reply and thanks for the quick reponse :)

Well, I want to iterate through each key value pairs of json object which is in array.
I am using iterator like this::

```
for (json::iterator it = jsonObj.begin(); it != jsonObj.end(); ++it) {
  for (auto & el: * it.items()) {
    std::cout << el.key() << " : " << el.value() << "\n";
  };
}
```

I am getting this error::
'items': is not a member of 'nlohmann::detail::iter_impl<nlohmann::basic_json<std::map,std::vector,std::st
ring,bool,int64_t,uint64_t,double,std::allocator,nlohmann::adl_serializer,std::vector<uint8_t,std::allocator<uint8_t>>>>'
'el': references must be initialized

Thanks in advance

**Reply by nlohmann** [reply]
Try this:

```cpp
for (json::iterator it = jsonObj.begin(); it != jsonObj.end(); ++it) {
  for (auto & el: it->items()) {
    std::cout << el.key() << " : " << el.value() << "\n";
  };
}
```

`items()` is a member function of type `json`, not of `json::iterator`.

**Reply by mayank1608** [reply]
@nlohmann Thanks Bro It is working fine now, got my mistake. Thanks for the help 👍
